JOSESwift/Sources/CryptoImplementation/AESGCM.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  AESGCM.swift
+//  JOSESwift
+//
+
+import Foundation
+import CryptoKit
+
+enum AESGCM {
+    typealias KeyType = Data
+
+    /// Encrypts a plain text using a given `AES.GCM` algorithm.
+    ///
+    /// - Parameters:
+    ///   - plaintext: The plain text to encrypt.
+    ///   - encryptionKey: The symmetric key.
+    ///   - initializationVector: The initial block.
+    ///   - additionalAuthenticatedData: The additional data block that is to be authenticated.
+    /// - Returns: The encryption context containing ciphertext, the authentication tag and the `IV`
+    /// - Throws: an error if any error occurs during encryption.
+    static func encrypt(
+        plaintext: Data,
+        encryptionKey: KeyType,
+        initializationVector: Data,
+        additionalAuthenticatedData: Data
+    ) throws -> ContentEncryptionContext {
+        let key = CryptoKit.SymmetricKey(data: encryptionKey)
+        let nonce = try CryptoKit.AES.GCM.Nonce(data: initializationVector)
+        let encrypted = try CryptoKit.AES.GCM.seal(plaintext, using: key, nonce: nonce, authenticating: additionalAuthenticatedData)
+        return ContentEncryptionContext(
+            ciphertext: encrypted.ciphertext,
+            authenticationTag: encrypted.tag,
+            initializationVector: encrypted.nonce.withUnsafeBytes({ Data(Array($0)) })
```

Comment:
Is 

``` swift
encrypted.nonce.withUnsafeBytes({ Data(Array($0)) })
```

the same as

``` swift
initializationVector
```

?

Reply:
Yes it is. This way all the context was built out of encrypted output. 
Changed it to `initializationVector` for readability now. Also involves fewer conversions.

Reply:
> This way all the context was built out of encrypted output.

That somehow sounds better to me I must admit. Just to double check, does it have any relevance at all if we use the incoming iv or the one from the encryption output?